Runtimes/Overlay/Android/clang/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+
+# FIXME: how do we determine the sysroot? `CMAKE_SYSROOT` does not contain the sysroot.
+file(CONFIGURE
+  OUTPUT android-ndk-overlay.yaml
+  CONTENT [[
+---
+version: 0
+case-sensitive: false
+use-external-names: false
+roots:
+  - name: "@CMAKE_ANDROID_NDK@/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include"
+    type: directory
+    contents:
+      - name: module.modulemap
+        type: file
+        external-contents: "@CMAKE_CURRENT_SOURCE_DIR@/android.modulemap"
+      - name: SwiftAndroidNDK.h
+        type: file
+        external-contents: "@CMAKE_CURRENT_SOURCE_DIR@/SwiftAndroidNDK.h"
+      - name: SwiftBionic.h
+        type: file
+        external-contents: "@CMAKE_CURRENT_SOURCE_DIR@/SwiftBionic.h"
+]]
+ESCAPE_QUOTES @ONLY NEWLINE_STYLE LF)
+
+add_library(SwiftAndroid INTERFACE)
+target_compile_options(SwiftAndroid INTERFACE
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xcc --sysroot=\"${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/windows-x86_64/sysroot\">"
```

Comment:
Any reason you're not using the new `-sysroot` flag you introduced in #72352 instead: is it not working?

Reply:
The `-sysroot` flag is part of the Swift interface, this is for the clang module (C interface). The time where this is being used, the compiler is not able to inject the headers/modules as they do not exist in any location that the compiler will search.

Reply:
I see, this is because the way CMake's built-in Swift support sets this up is fundamentally different? Because the current bespoke CMake config in `stdlib/`, that doesn't use CMake's built-in Swift support, doesn't need to pass in these two flags.

Reply:
From what I can tell, it's because we're overlaying the module files over the Android NDK rather than copying them into the compiler resource directory like the old system does in [`stdlib/public/Platform/CMakeLists.txt`](https://github.com/swiftlang/swift/blob/97b249bd117f72978fb87be1c376c6dac2c731dd/stdlib/public/Platform/CMakeLists.txt#L409-L475)

Reply:
Exactly - we aren't copying things into the location and then using it, we are using it directly and then installing at the end to the proper location.